python/lsst/daf/persistence/butlerLocation.py

```diff
@@ -27,9 +27,65 @@
 
 import lsst.daf.base as dafBase
 
+from collections import namedtuple
+from past.builtins import basestring
 import yaml
 
-from . import iterify
+from . import iterify, doImport
+
+
+class ButlerComposite(object):
+    ComponentInfo = namedtuple('ComponentInfo', 'datasetType')
+
+    def __init__(self, assembler, disassembler, python, dataId, mapper):
+        """Initializer
+
+        Parameters
+        ----------
+        assembler : function object
+            Function object or importable string to a function object that can be called with the assembler
+            signature: (dataId, componentDict, cls).
+        disassembler : function object
+            Function object or importable string to a function object that can be called with the disassembler
+            signature: (object, dataId, componentDict).
+        python : class object
+            A python class object or importable string to a class object that can be used by the assembler to
+            instantiate an object to be returned.
+        dataId : dict or DataId
+            The dataId that is used to look up components.
+        mapper : Mapper instance
+            A reference to the mapper that created this ButlerComposite object.
+        """
```

Comment:
I think docstrings for `__init__` functions don't actually work as you might expect in Python (they don't appear when you ask for on-line help on the type).  It's standard to instead document the constructor and its parameters in the class docstring.  More on that [here](https://developer.lsst.io/docs/py_docs.html#py-docstring-class-structure).